### ISSUE TITLE
Permission denied on `composer update`

### ISSUE THREAD

**Issue by LimeBlast** [fideloper/Vaprobash#174]
I've got a feeling this might be a Windows only issue (or more specifically, a non-NFS issue), but after successfully provisioning the VM, I SSH into the server and run `composer update` from the `/vagrant` folder, at which point I get the following error:

```
Loading composer repositories with package information
Permission denied
https://packagist.org could not be fully loaded, package information was loaded from the local cache and may be out of date
```

I've done a bit of searching and found that this is happening because the `/home/vagrant/.composer` folder is owned by root (seeing as the first time the `composer` command is run, it is as the `root` user) but I'm trying to run `composer update` as `vagrant`.

The fix for this is easy, you simply need to run:

```
sudo chown -R vagrant: /home/vagrant/.composer
```

and the permissions of the folder are set correctly - so I'm guessing this command could be added to `composer.sh` towards the end of the file - but I don't know what sort of effect this would have on the NFS users.

Any ideas?

**Comment by generalredneck** [comment]
just a quick heads up... If you come across this issue (as I had) and you are running composer on a windows box inside windows (not from inside your VM)... you need to give your commandline session Administration rights to change permissions. The simplest way to do so is to shift click when you open Commandline... or rightclick and select "Run as Administrator" when using the shortcut.

**Comment by therobyouknow** [comment]
I don't understand why composer had to change permissions for the following reasons:
a) if they are already sufficient, e.g. 777 for directories, 666 for files - so why change them?
b) on a developer's local environment, requiring permissions is not necessary for security measures

**Comment by therobyouknow** [comment]
I've found that even if the owner of composer files is vagrant and even if the permissions are 777 for directories and 666 for files, composer fails to set permissions. This seems to be simply not possible on an NFS share inside a VM.

The workarounds are:
- prefix with sudo
- run composer on the host instead

I raised an issue about why composer has to change permissions when they should be sufficient enough, here:
https://github.com/composer/composer/issues/7662

In that same issue, I detailed the thinking behind the above workarounds.
